droid-container/src/main/java/uk/gov/nationalarchives/droid/container/ContainerSignatureMatch.java

```diff
@@ -165,7 +165,7 @@ public void matchBinaryContent(String entryName, ByteReader content) {
             Map<String, ContainerFile> sigFiles = signature.getFiles();
             InternalSignatureCollection binSigs = sigFiles.get(entryName).getCompiledBinarySignatures();
             if (binSigs != null) {
-                matched = binSigs.getMatchingSignatures(content, maxBytesToScan).size() > 0;
+                matched = binSigs.getMatchingSignatures(content, maxBytesToScan, 1).size() > 0;

```

Comment:
Why `1` here instead of `-1` (which would have been the previous default)?

Reply:
It is a speed optimization, using early termination: The overall check is whether there is any match at all (the `> 0` part). As soon as there is 1 hit, that check is satisfied and there is no need to look for further hits.